e2e/nomostest/registry_server.go

```diff
@@ -97,31 +97,43 @@ func setupRegistry(nt *NT) error {
 	return nil
 }
 
-// setupRegistryClient handles registry authentication.
-// This is necessary on each test because gcloud auth tokens expire after 1h and
-// some test suites take longer than 1h.
+// setupRegistryClient handles registry authentication and port-forwarding.
+//
+// Port-forwarding is configured for local registries so that if the registry
+// pod crashes or is rescheduled the port-forwarding will be restarted after it
+// becomes healthy again. Then the client will re-login and re-push all the
+// images and charts that were previously pushed during the same test.
+//
+// For non-local registries, only login, logout, and cleanup are performed.
+//
+// Helm provider is only configured if RequireHelmProvider is true.
+// OCI provider is only configured if RequireOCIProvider is true.
 func setupRegistryClient(nt *NT, opts *ntopts.New) error {
-	nt.T.Cleanup(func() {
-		if opts.RequireHelmProvider || opts.RequireLocalHelmProvider {
-			if err := nt.HelmProvider.Logout(); err != nil {
-				nt.T.Error(err)
-			}
+	// For local registries, set up port-forwarding to the in-cluster registry server.
+	// portForwardRegistryServer handles login and logout too.
+	if opts.RequireOCIProvider && *e2e.OCIProvider == e2e.Local || opts.RequireHelmProvider && *e2e.HelmProvider == e2e.Local {
```

Comment:
Can we rename `RequireOCIProvider`/`RequireHelmProvider` to differentiate them from the other `Require*` options? The other `Require` options indicate to skip the test (e.g. `RequireLocalHelmProvider`), which is functionally quite different from this option.

I'm thinking something along the lines of
- `IsHelmTest`/`IsOCITest`
- `UsesHelmProvider`/`UsesOCIProvider`
- etc

Reply:
Yeah, I get that it's a little confusing. My thinking is that the test defines its requirements, and the test infra either injects it or skips it.

Reply:
That's fair, works for me